### DISCUSSION TITLE
Constraint for minimum pickup demand depending on vehicle capacity

### DISCUSSION THREAD

**Discussion by alti3** [google/or-tools#3197] [Routing (and legacy CP) questions] [answered]
In a CVRP with Pickup/Delivery constraints, is it possible to create a constraint that effectively acts the opposite of the capacity constraint.

The problem I'm solving involves several vehicle types with different capacities (Delivery scooter, Car, Van, Truck...etc).
As with any CVRP, each pickup location has a `Demand` representing item size/weight.

I want to prevent the solver from assigning small item pickups to unnecessarily large vehicles (Vans, Trucks).

I thought maybe I can add a constraint that sets a minimum pickup demand for each vehicle.

Is this possible with OR-Tools?

**Comment by vuongdanghuy** [comment, marked as answer]
I don't think it's possible to set a minimum pickup demand. But if you want prohibit some vehicles from visiting a node, try something like `SetAllowedVehiclesForIndex`.

Hope this can help.

**Reply by alti3** [reply]
simple yet elegant solution
thanks

**Comment by jmarca** [comment]
Well if you know a priori what you don’t want to be matched, then just prevent those by saying the node can’t be served by those vehicles.

**Comment by Mizux** [comment]
An other soft solution (not tested) would be:
* Create a counter dimension per package type size: small, middle, large.
* each time a vehicle visit a pickup increase the counter accordingly
* For large vehicle if you don't want to pickup small package add a soft constraint on the counter to be zero.
```py
vehicle_large_end = routing.End(vehicle_large_index)
small_package_dimension.SetCumulVarSoftUpperBound(
  vehicle_large_end, # index of the vehicle end node
  0, # upper_bound here 0 we don't want any small package pickup performed by this large vehicle
  42_000 # penalty coefficient aka penalty cost per pickup visited
)
```
If this large vehicle visit the small package pickup location, then the solver (objective value) will pay the penalty cost.
So it will give incentive the the solver to avoid it unless 42_000 cost is lower than not performing the delivery (if node optional with larger disjunction penalty) or the use of smaller vehicle will increase the overall cost above this value (e.g. the vehicle arc cost increase is higher than asking the large vehicle to visit it)

ref: 
https://github.com/google/or-tools/blob/49b6301e1e1e231d654d79b6032e79809868a70e/ortools/constraint_solver/routing.h#L2740-L2749
